packages/common/test/utils.spec.ts

```diff
@@ -108,4 +96,14 @@ describe('[Utils/Parse]', () => {
       'should parse correct address',
     )
   })
+  it('should add mergeNetSplitBlock if not present when Shanghai is present', () => {
+    const genesisJSON = postMergeData
```

Comment:
This works for test data (hive), but it will not work if someone explicitly imports mainnet genesis as "custom chain". Then we will end up with wrong fork hashes :thinking:

Reply:
As you and I discussed, mainnet it doesn't seem to have a mergeNetsplitBlock but it just be there somewhere. I honestly don't think this is an issue since why would somebody ever run mainnet from a geth genesis file?